### DISCUSSION TITLE
Building on Ubuntu 22.04

### DISCUSSION THREAD

**Discussion by dts-1990** [hardening/qfreerdp_platform#16] [Q&A] [answered]
Hi, first of all thank you for sharing this plugin! I'm looking for this to replace VNC due to performance issue when sharing over web/internet, VNC is quite slow because it is basically Screen capturing. 

I'm trying to compile this plugin on Ubuntu 22.04, all dependencies mentioned on README installed, but I face this when running meson:

`meson.build:35:17: ERROR: Dependency "winpr3" not found, tried pkgconfig and cmake
`

Tried to find winpr3 but looks like it is only available on Ubuntu 24.04. Do you have any ideas/suggestion to overcome this err?
Thanks

**Comment by hardening** [comment, marked as answer]
Well install FreeRDP3 and point `PKG_CONFIG_PATH` to the right location

**Reply by dts-1990** [reply]
Thank you, I've built the lib successfully now and installed it to /usr/local/lib.
But running my program it exits with err that no rdp plugin:

`export QT_PLUGIN_PATH=/usr/local/lib
myprogram -platform freerdp
Warning: Could not find the Qt platform plugin "freerdp" in ""
Fatal: This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, webgl, xcb.

Aborted (core dumped)`

What am I missing?

**Reply by hardening** [reply]
That's your `QT_PLUGIN_PATH` variable that needs to be adjusted, I'd say that it should be something like `/usr/local/lib/x86_64-linux-gnu/qt6/plugins`.

**Reply by dts-1990** [reply]
> That's your `QT_PLUGIN_PATH` variable that needs to be adjusted, I'd say that it should be something like `/usr/local/lib/x86_64-linux-gnu/qt6/plugins`.

:( Still no luck with: 
`export QT_PLUGIN_PATH=/usr/local/lib/x86_64-linux-gnu/qt5/plugins`
or 
`export QT_PLUGIN_PATH=/usr/local/lib/x86_64-linux-gnu/qt5/plugins/platforms`

Even I see the lib installed at /usr/local/lib/x86_64-linux-gnu/qt5/plugins:
`ll /usr/local/lib/x86_64-linux-gnu/qt5/plugins/platforms/
total 9824
drwxr-xr-x 2 root root     4096 Oct 24 17:30 ./
drwxr-xr-x 3 root root     4096 Oct 24 17:30 ../
-rwxr-xr-x 1 root root 10047704 Oct 24 17:30 libqfreerdp.so*
`

NOTE: Actually, I configured meson to build with qt5 since my app also built with qt5.

**Reply by hardening** [reply]
Try setting `QT_DEBUG_PLUGINS=1` and you'll have more debug messages on the module loading.

**Reply by dts-1990** [reply]
Nice! Found that, I set wrong LD_LIBRARY_PATH, that's stupid on me.  Thank you very much

Now I can expose my app through RDP, but fail to connect because of missing SSL key. Let me try to find a way out on that
